rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+# CSV Preview
+
+[govuk-content-schemas]: https://github.com/alphagov/govuk-content-schemas
+[government-prefix]: https://trello.com/c/74KTVXBs/52-whitehall-relies-on-prefix-government-and-world-routes
+[router]: https://github.com/alphagov/router#backend-handler
+
+## Summary
+
+Move the CSV Preview feature out of Whitehall and into Government Frontend, so that it can be used for documents published by other apps, like Content Publisher. The new implementation will work by rendering a content item in a new `attachment_preview` format, which represents an attachment to preview.
+
+## Problem
+
+A user viewing a Whitehall document on GOV.UK can click a "View Online" link for any of its CSV file attachments. This takes the user to a new page, currently rendered by Whitehall. The page contains:
+
+   - A 'header' section with details of the parent document (and navigation links)
+   - A 'body' section with a preview of the content of the CSV attachment
+
+This CSV Preview feature only works for CSV file attachments uploaded through Whitehall. We need it to work for CSV attachments published through Content Publisher, which is superseding Whitehall.
+
+## Proposal
+
+### Step 1
+
+Create a new `attachment_preview` format in [govuk-content-schemas][] to represent the preview page for an attachment. A publishing app will publish content items in this format for each CSV attachment to preview.
+
+```
+# formats/attachment_preview.jsonnet
+(import "shared/default_format.jsonnet") + {
+  document_type: "attachment_preview",
+  attachment: {
```

Comment:
Should this belong in the details hash?

Reply:
I was wondering about that. It seemed a bit weird/verbose, since the attachment pretty much _is_ the details. Are you thinking it's more conventional?

Reply:
I think we can continue a discussion about this as part of implementation. The intention here is really just to show the data would be there is some format.